libs/src/api/Track.ts

```diff
@@ -411,21 +419,36 @@ export class Track extends Base {
       phase = phases.ADDING_TRACK
 
       // Write metadata to chain
-      const multihashDecoded = Utils.decodeMultihash(metadataMultihash)
-      const { txReceipt, trackId } =
-        await this.contracts.TrackFactoryClient.addTrack(
+      let txReceipt: TransactionReceipt
+      let trackId: number
+      if (useEntityManager) {
+        trackId = Track.generateTrackId()
+        const response = await this.contracts.EntityManagerClient!.manageEntity(
```

Comment:
for subsequent changes should we modify existing files like this? maybe entityManager.ts functions should move over to playlist.